src/EditorFeatures/CSharpTest/UseRecursivePatterns/UseRecursivePatternsTests.cs

```diff
@@ -0,0 +1,515 @@
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test6()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" [||]&& !type.IsClass;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", IsClass: false
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test7()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.ContainingSymbol is var declContainer &&
+            declContainer.Kind == SymbolKind.Namespace &&
+            declContainer.Name == ""System"" [||]&&
+            (declContainer.ContainingSymbol as NamespaceSymbol)?.IsGlobalNamespace == true;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            ContainingSymbol:
+            {
+                ContainingSymbol: NamespaceSymbol
+                {
+                    IsGlobalNamespace: true
+                }
+
+                , Name: ""System"", Kind: SymbolKind.Namespace
+            }
+
+            declContainer
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test8()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return (declContainer.ContainingSymbol as NamespaceSymbol)?.IsGlobalNamespace [||]== true;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return declContainer is
+        {
+            ContainingSymbol: NamespaceSymbol
+            {
+                IsGlobalNamespace: true
+            }
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test9()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch (node)
+        {
+            case String s [||]when s.Length == 0:
+                break;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (node)
+        {
+            case String
+            {
+                Length: 0
+            }
+
+            s:
+                break;
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test10()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch (node)
+        {
+            case String s [||]when s.Length == 0 && b:
+                break;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (node)
+        {
+            case String
+            {
+                Length: 0
+            }
+
+            s when b:
+                break;
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test16()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch (node)
+        {
+            case String s [||]when b && s.Length is int && c:
```

Comment:
Is it ok to reorder these expressions (potentially with sideeffects) to at least offer the fix? User can undo if the result is not desirable.